### PR TITLE
remove code duplication

### LINKED ISSUE #105798: Reduce code duplication in test/e2e/storage/testsuites/snapshottable.go
### What would you like to be added?

The changes to support snapshot tests for Generic Ephemeral Volumes https://github.com/kubernetes/kubernetes/pull/105659 generated some duplicated code in the file `test/e2e/storage/testsuites/snapshottable.go` as commented here: https://github.com/kubernetes/kubernetes/pull/105659#discussion_r732587703, it'd be great to create functions that are reused in both `(ephemeral)` and `(persistent)` tests.

Things that are needed:

- A kubernetes dev cluster (I follow [this guide](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/windows/README-GCE-Windows-kube-up.md) to create a GCE cluster)
- Knowledge about how to [run e2e tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md)

/sig storage
/area testing
/good-first-issue

### Why is this needed?

Improve readability of the storage VolumeSnapshot tests

### PR DESCRIPTION
**Pull request by acumino**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Improve readability of the storage VolumeSnapshot tests
#### Which issue(s) this PR fixes:

Fixes #105798

#### Special notes for your reviewer:
CC: @mauriciopoppe
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/testsuites/snapshottable.go [hunk 1]
```diff
@@ -172,7 +173,7 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 		})
 
 		ginkgo.Context("", func() {
-			ginkgo.It("should check snapshot fields, check restore correctly works, check deletion (ephemeral)", func() {
+			ginkgo.It("should check snapshot fields, check restore works correctly, check deletion (ephemeral)", func() {
```
Comment (mauriciopoppe) [inline test/e2e/storage/testsuites/snapshottable.go:176]: Even though this looks like a simple wording change I believe that we have some internal tests that will break if make this change, I think this change can be done later.

**Review by mauriciopoppe — COMMENTED** [review]
/ok-to-test

Thanks for the PR! We've just found out that there are a few bugs that we recently introduced in this refactor https://github.com/kubernetes/kubernetes/commit/c0bdf149425d4b369fce09cd934e7734e12b2cd0, I'll submit a followup PR attempting to fix some of those bugs and then we can add this one on top :)

Bug: we no longer wait for the pod to be completely deleted before taking the snapshot like we used to.

File: test/e2e/storage/testsuites/snapshottable.go (hunk 1 above)
Comment (acumino) [reply test/e2e/storage/testsuites/snapshottable.go:176]: Yes, these wording changes are not important for now.

**Comment by acumino** [comment]
@mauriciopoppe now #107173 has been merged probably we can move ahead with this PR?

File: test/e2e/storage/testsuites/snapshottable.go (hunk 1 above)
Comment (mauriciopoppe) [reply test/e2e/storage/testsuites/snapshottable.go:173]: please undo this line in both tests

**Review by mauriciopoppe — COMMENTED** [review]
adding @jingxu97 and @msau42 to review the changes too

**Comment by mauriciopoppe** [comment]
please rebase the changes in a single commit too :)

/cc @jingxu97

**Comment by mauriciopoppe** [comment]
/assign @msau42 

This is a small refactor on the snapshottable file to reduce duplication between the two versions of the test, your feedback would be appreciated :)

**Comment by msau42** [comment]
/approve
Will let @mauriciopoppe give final lgtm

File: test/e2e/storage/testsuites/snapshottable.go [hunk 2]
```diff
@@ -222,32 +221,13 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
...
-
-				// Get the bound snapshotContent
-				snapshotStatus := vs.Object["status"].(map[string]interface{})
-				snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
-				vscontent, err := dc.Resource(storageutils.SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
-				framework.ExpectNoError(err)
-
-				snapshotContentSpec := vscontent.Object["spec"].(map[string]interface{})
-				volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+				vs := sr.Vs
```
Comment (mauriciopoppe) [inline test/e2e/storage/testsuites/snapshottable.go:224]: in the previous code `vs` is updated for the rest of the function with the latest version from the API server

```
vs, err = dc.Resource(storageutils.SnapshotGVR).Namespace(vs.GetNamespace()).Get(context.TODO(), vs.GetName(), metav1.GetOptions{})
```

in the refactor that update only happens within the helper function, luckily, the properties of `vs` in this function that came from the update aren't used so it should work

**Review by mauriciopoppe — COMMENTED** [review]
/lgtm

I think this test doesn't run in the kubernetes presubmit checks, however I'll check a CI run in https://testgrid.k8s.io/provider-gcp-compute-persistent-disk-csi-driver#Kubernetes%20Master%20Driver%20Latest, thank you for your contribution!

**Comment by msau42** [comment]
@mauriciopoppe did you check pull-kubernetes-e2e-gce-storage-snapshot

**Comment by mauriciopoppe** [comment]
@msau42 I just checked it and both the ephemeral and persistent cases are passing, cool I didn't know about that setup with csi-hostpath and pd csi

**Pull request merged**